Sources/tuist/main.swift

```diff
@@ -5,6 +5,9 @@ import TuistLoader
 import TuistSupport
 
 if CommandLine.arguments.contains("--verbose") { try? ProcessEnv.setVar(Constants.EnvironmentVariables.verbose, value: "true") }
+if CommandLine.arguments.contains("--generate-completion-script") {
```

Comment:
I think having this logic in `tuistenv/main.swift` should be sufficient because it's the process where the `Using version 2.3.2 defined at /Users/pepibumur/Downloads/.tuist-version` message is printed.

Reply:
It is not sufficient because the `logLevel` is determined [here](https://github.com/tuist/tuist/pull/3804/files/b59038f8fc791c5cd68b177de79735a6f5687b2f#diff-1a62a9a8037ea6a1e8ca5df02c2401c43f3abbc04623f4ffbcce01ffa050bcdeR69) based on the `silent` environment variable. 

Alternatively, we could dynamically change the `logLevel` for the `tuistenv`'s `logger`. However, if we add _any_ log in `tuist` (apart from generating the script), the script will break again. I would be fine with this limitation if we could write an acceptance test to make sure no such regression occurs in the future but I'm not sure how feasible that is.

Reply:
I lean in that case towards doing that only for `tuistenv` and add an acceptance test.

Reply:
@pepicrft So are you suggesting to revert the last commit (https://github.com/tuist/tuist/pull/3804/commits/b59038f8fc791c5cd68b177de79735a6f5687b2f)? Then the log level would be changed only in `tuistenv`.

Will it be possible to use custon-built `tuistenv` and make it run custom-built `tuist` in an acceptance test?

Reply:
> @pepicrft So are you suggesting to revert the last commit (b59038f)? Then the log level would be changed only in tuistenv.

Yes

> Will it be possible to use custom-built tuistenv and make it run custom-built tuist in an acceptance test?

Are you thinking about doing that in order to be able to test the scenario?

Reply:
> Are you thinking about doing that in order to be able to test the scenario?
@pepicrft  Yes. In order to test this scenario we would need to use both binaries produced by the build process (`tuist` and `tuistenv`).

Reply:
What do you mean by custom-built `tuist`? We could also just assert what the first line of the generated script is. The first line is currently `#compdef tuist` which is imho unlikely to change. Of course, it does not test whether the completion actually works but will definitely catch the regression of the script having additional log output from tuist.

Reply:
> What do you mean by custom-built tuist?

I mean the version of `tuist` that has just been built for the purpose of testing.

Maybe I'm missing something obvious but I haven't find a way to fully test the solution without manually installing the new `tuistenv` build in `/usr/local/bin` and `tuist` in `~/.tuist/Versions`.
If I run `tuist --generate-completion-script`, `tuistenv` binary is not run at all - so the unwanted log message is not outputted. On the other hand, running `tuistenv --generate-completion-script` with `.tuist_version` file present in the current directory, will trigger fetch of the given version to `~/.tuist/Versions` and this released version will will be used instead of the version that should be tested. 

So I know how to assert, but I don't know how to set up environment for this test.

Reply:
> On the other hand, running tuistenv --generate-completion-script with .tuist_version file present in the current directory, will trigger fetch of the given version to ~/.tuist/Versions and this released version will will be used instead of the version that should be tested.

True, I think there is not a way how to achieve that as of now. What you could do is to run the `tuist` binary path with `TUIST_CONFIG_BINARY_PATH=@tuist @tuistenv --generate-completion-script`. The env variable does not exist, yet, but will when [plugins 2.0 are merged](https://github.com/tuist/tuist/pull/3492/files#diff-e7ba6c0d31dcd07231e606f90cf522bfdd2efe88c92a3fce22e7fe078972ba4aR81). You could then check for the variable [here](https://github.com/tuist/tuist/blob/main/Sources/TuistEnvKit/Commands/CommandRunner.swift#L85) in the `CommandRunner` instead of searching for the highest local version/version specified in `.tuist-version`

Reply:
I see that the PR has been merged. So I guess we can add that test once `TUIST_CONFIG_BINARY_PATH` is available alongside Plugins 2.0.